### DISCUSSION TITLE
How do I enable AlphaToCoverage?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#12199] [Help Wanted]
In StandardPBR_ShaderEnable.lua, I added the following code:

    local forwardPassRenderStates = forwardPass:GetRenderStatesOverride()

    if opacityMode == OpacityMode_Cutout then

        forwardPassRenderStates:SetAlphaToCoverageEnabled(true)

    end

This results in grey rendering.

**Comment by ohmaya** [comment]
The Cutout object is drawn (but no pixel get written) in Subsurface pass even if the material has Subsurface Scattering off.

By turning on Subsurface Scattering, the Cutout object is rendered, however, **no AlphaToCoverage effect is shown**.

The shader code resulting in the grey look:
EnhancedSurface_ForwardPass.azsli
lightingOutput.m_diffuseColor.w = factorAndQuality * (o_enableSubsurfaceScattering ? 1.0 : **-1.0**);

**Comment by santorac** [comment]
I'm not sure off the top of my head. I haven't tried alpha-to-coverage before. A couple things you might try:
- Use RenderDoc, PIX, or NSight to do a GPU capture and see if that reveals anything.
- Try using StandardPBR instead of EnhancedPBR to simplify things and narrow down what's going wrong.
- I just noticed there is a AlphaToCoverage sample in [AtomSampleViewer](https://github.com/o3de/o3de-atom-sampleviewer). You could investigate there and look for difference between the sample and your work.
